packages/app/src/pages/Specs/Index.vue

```diff
@@ -1,14 +1,24 @@
 <template>
   <div v-if="query.data.value">
+    <CreateSpecModal
+      v-if="query.data.value.currentProject?.currentTestingType"
+      :key="generator"
+      :initial-generator="generator"
+      :show="modalIsShown"
+      :gql="query.data.value"
+      @close="closeCreateSpecModal"
+    />
```

Comment:
This is the largest non-test change in the PR. The Index page switches between the NoSpecsPage and SpecsList components as users create them; when each component maintains its own CreateSpecModal state, the modal cannot remain open during that transition and the dialog closes unexpectedly.

I lifted that state up to mitigate this, feedback on implementation welcome. I'm curious if there's a less boilerplate-y way to do this with Vue.